integration/integration.go

```diff
@@ -17,27 +17,41 @@ type Integration struct {
 	Name               string    `json:"name"`
 	ProtocolVersion    string    `json:"protocol_version"`
 	IntegrationVersion string    `json:"integration_version"`
-	Data               []*Entity `json:"data"`
+	Entities           []*Entity `json:"data"`
 	prettyOutput       bool
 	writer             io.Writer
 }
 
 // New creates new integration with sane default values.
-func New(name, version string) (*Integration, error) {
-	return NewBuilder(name, version).Build()
+func New(name, version string, args interface{}) (*Integration, error) {
+	return NewBuilder(name, version).ParsedArguments(args).Build()
 }
 
-// NewWithArgs creates new integration with given arguments and sane default values.
-func NewWithArgs(name, version string, args interface{}) (*Integration, error) {
-	return NewBuilder(name, version).ParsedArguments(args).Build()
+// DefaultEntity retrieves default entity to monitorize.
+func (i *Integration) DefaultEntity() *Entity {
+	i.locker.Lock()
+	defer i.locker.Unlock()
+
+	for _, e := range i.Entities {
+		if e.IsDefaultEntity() {
+			return e
+		}
+	}
```

Comment:
This piece of code may be a bottleneck (specially if synchronized). I would implement it the next way:

-  Add to the `Inventory` struct a private pointer to the default entity (which will not be serialized into JSON)
- If the `defaultEntity` pointer is nil, create the new default property, append it to the `i.Entities` and also point the `defaultEntity` pointer to it.
- Otherwise, return directly the `defaultEntity` pointer.